### ISSUE TITLE
Internal error: !NULL destructor called

### ISSUE THREAD

**Issue by slayoo** [gnudatalanguage/gdl#220]
As reported by @maynardGK on SF.net (https://sourceforge.net/p/gnudatalanguage/bugs/698/):

```
GDL>  tst1 = list()
GDL>  help,tst1.toarray()
<Expression>    UNDEFINED = !NULL
GDL> i=where( tstcmp ne tst1.toarray(), nc0)
Internal error: !NULL destructor called.
Save your work and restart GDL (GDL is still functional, but !NULL will not work anymore).
Please report at http://sourceforge.net/tracker/?group_id=97659&atid=618683
```

**Comment by jtappin** [comment]
Looks like just about any attempt to access an unassigned list element triggers this, e.g.:
````
GDL> ll = list(len=4)
GDL> ll[0]=45
GDL> ll[2]=76.4
GDL> for j = 0, 3 do print, ll[j] eq !null
   0
Internal error: !NULL destructor called.
Save your work and restart GDL (GDL is still functional, but !NULL will not work anymore).
Please report at https://github.com/gnudatalanguage/gdl/issues
   1
   0
Internal error: !NULL destructor called.
Save your work and restart GDL (GDL is still functional, but !NULL will not work anymore).
Please report at https://github.com/gnudatalanguage/gdl/issues
   1
````
Running GDL 1.0.0-rc3.

**Comment by GillesDuvert** [comment]
I believe !NULL will be useable (in the limit of !NULL being useable in GDL, so, not much) despite the warning. TBC.
There are now 4 !NULL tagged issues, showing the implementation of !NULL in the interpreter is very unsatisfactory.

**Comment by GillesDuvert** [comment]
closed by #1174
